### PR TITLE
Indicator LEDs as config

### PR DESCRIPTION
**Pull request by zvecr**
## Types of Changes

- [x] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [ ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [ ] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by drashna — APPROVED** [review]

**Comment by drashna** [comment]
I like this change, a lot. 

Only concern I have is if the pins aren't wired for output.  But it's set weakly, shouldn't be too much of a concern, really.

File: docs/feature_led_indicators.md [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+To configure the indicators, `#define` these in your `config.h`:
+
+|Define               |Default      |Description                                |
+|---------------------|-------------|-------------------------------------------|
+|`LED_NUM_LOCK_PIN`   |*Not defined*|The pin that controls the `Num Lock` LED   |
+|`LED_CAPS_LOCK_PIN`  |*Not defined*|The pin that controls the `Caps Lock` LED  |
+|`LED_SCROLL_LOCK_PIN`|*Not defined*|The pin that controls the `Scroll Lock` LED|
+|`LED_COMPOSE_PIN`    |*Not defined*|The pin that controls the `Compose` LED    |
+|`LED_KANA_PIN`       |*Not defined*|The pin that controls the `Kana` LED       |
+|`LED_PIN_ON_STATE`   |`0`          |The state of the indicator pins when the LED is "on" - `1` for high, `0` for low|
```
Comment (fauxpark) [inline docs/feature_led_indicators.md:34]: I kind of feel like 1 should be the default as it is for Backlight.

Comment (zvecr) [reply docs/feature_led_indicators.md:34]: I just went with what the example in the docs had, easy enough to change. I guess we can try and sample the repo for the most popular scheme and target that as the default?

Comment (zvecr) [reply docs/feature_led_indicators.md:34]: From a terrible greppy count, it seems to be fairly evenly split. I don't think there is much harm in aligning with backlight.

**Review by nooges — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Pull request merged**